### PR TITLE
feat: generate node api in TypeScript

### LINKED ISSUE #1172: Drop custom JSON parser and use `int-to-str` option instead
https://github.com/aeternity/aeternity/blob/v5.9.1/docs/release-notes/RELEASE-NOTES-5.9.1.md

### LINKED ISSUE #1216: Statically generate node and compiler APIs from Swagger files
Like it is done in go-sdk. This would help to
- create a TypeScript version of it (needed for #1215)
- make sdk to init in sync (#762)

### PR DESCRIPTION
**Pull request by davidyuk**
fixes #1172
fixes #1216 
blocked by https://github.com/aeternity/aeternity/issues/3668 https://github.com/aeternity/aeternity/pull/3890
requested support in https://github.com/Azure/autorest.typescript/issues/1053
found issues: https://github.com/aeternity/aeternity/issues/3892 https://github.com/aeternity/aeternity/issues/3891

### Comparison of packages to use

| Name | Activity | Comment |
|---|---|---|
| autorest | 28k downloads, 3.4k stars, active development | Microsoft, converts key to camelCase, multiple languages support, can't define a custom type, acceptable bundle size |
| openapi-typescript-codegen | 13k downloads, 359 stars, 3m ago | Generated structure looks clear, converts key to camelCase, template-based |
| swagger-typescript-api | 10k downloads (progressive), 495 stars, active development | template-based |
| **Would be used unlikely** |
| openapi-typescript | 36k downloads (progressive), 941 stars, active development | generates only typings |
| dtsgenerator | 23k downloads, 369 stars, 1m ago | generates only typings |
| swagger-codegen | 13k stars, active development | written in Java, generates tests |
| NSwag | 4.4k stars, active development | written in C# |
| **Not an option** |
| swagger-typescript-codegen | 7k downloads, 126 stars, active development | no cli, only Swagger 2 specs are supported |
| swagger-js-codegen | 3k downloads, 631 stars, 2y ago | no cli |
| swagger-ts-client | 0.6k downloads, 25 stars, 3y ago | |

I think `autorest` with a typescript plugin is a good way to go except that we need some dirty workarounds to make it return/accept BigNumbers.

### CODE REVIEW

**Review by thepiwo — COMMENTED** [review]
whats the state of this?

**Comment by davidyuk** [comment]
I did the same for the compiler in #1456, proposing to merge that and then finish this one

File: src/chain.ts [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+/**
+ * Submit a signed transaction for mining
+ * @param tx Transaction to submit
+ * @param options
+ * @param options.onNode Node to use
+ * @param options.onAccount Account to use
+ * @param options.verify Verify transaction before sending
+ * @param options.waitMined Ensure that transaction get into block
+ * @param options.confirm Number of micro blocks that should be mined after tx get included
```
Comment (davidyuk) [inline src/chain.ts:75]: I'm proposing to do docs this way: https://tsdoc.org/

**Review by subhod-i — APPROVED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   node:
-    image: aeternity/aeternity:${NODE_TAG}
+    image: davidyuk/aeternity:${NODE_TAG}-for-sdk
```
[the remark this answers is not in the document]
Comment (davidyuk) [reply docker-compose.yml:4]: yea, it is basically https://github.com/aeternity/aeternity/pull/3890

**Pull request merged**